### PR TITLE
docs: adding contributing page

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
Adding contributing section to the docs

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+## Pull requests
+
+We love to see pull requests!
+
+We are using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+The two most important types: "fix" and "feat", would result in the new version of the `Pytest-splunk-addon` once merged.
+
+To do the changes you think are needed, run the previous steps (build / test / linting / documentation).
+After you create a PR, all the needed reviewers will be added automatically by GitHub.
```
Comment (kdoroszko-splunk) [inline docs/contributing.md:98]: Can we be more specific here and say towards which branch PR should be raised (`develop`)? Or add the information to use `squash merge` while merging the PR?

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+You can create an [issue](https://github.com/splunk/pytest-splunk-addon/issues) on GitHub.
+
+## Pull requests
+
+We love to see pull requests!
+
+We are using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+The two most important types: "fix" and "feat", would result in the new version of the `Pytest-splunk-addon` once merged.
+
+To do the changes you think are needed, run the previous steps (build / test / linting / documentation).
```
Comment (mkolasinski-splunk) [inline docs/contributing.md:97]: TBH I don't get this point.

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+Prerequisites:
+
+- Docker version: 25.0.3
+- Docker Compose version: v2.24.6-desktop.1
+
+Steps to run e2e tests locally:
+
+```
+git clone --recurse-submodules -j8 git@github.com:splunk/pytest-splunk-addon.git
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply docs/contributing.md:36]: I think we need cim app installed in splunk, but not sure we can take up a task to clean that up finally.

File: docs/contributing.md (hunk 2 above)
Comment (harshilgajera-crest) [reply docs/contributing.md:97]: Well, Ref for this was taken from https://github.com/splunk/addonfactory-ucc-generator/blob/develop/docs/contributing.md as this is the only tool that has contributing section

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+## Documentation changes
+
+Documentation changes are also welcome!
+
+To verify changes locally:
+
+```bash
+poetry run mkdocs serve -a localhost:8001
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply docs/contributing.md:83]: So user would not need to worry about that as themes and plugins would get installed once they install PSA.

> Should we put them in pyproject.toml in dedicated group like [tool.poetry.group.docs]?

Sure we can take that up

File: docs/contributing.md (hunk 1 above)
Comment (harshilgajera-crest) [reply docs/contributing.md]: Done

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
+# Contributing Guidelines
+
+
+# Prerequisites:
+
+- Poetry 1.5.1. [Installation guide](https://python-poetry.org/docs/#installing-with-the-official-installer)
+
+# Installation
+
+You can install "pytest-splunk-addon" via [pip] from [PyPI]:
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply docs/contributing.md:10]: Done

File: docs/contributing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkolasinski-splunk) [reply docs/contributing.md:41]: Created Jira to explore further: https://splunk.atlassian.net/browse/ADDON-74198

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**